radis/lbl/loader.py

```diff
@@ -2260,14 +2264,37 @@ def get_partition_function_calculator(self, molecule, isotope, elec_state):
         isotope: int
         elec_state: str
         """
-
-        parsum = self.parsum_calc[molecule][isotope][elec_state]
-
+        try:
+            parsum = self.parsum_calc[molecule][isotope][elec_state]
+        except KeyError as err:
+            raise KeyError(
+                "Error while Retrieving Partition Function Calculator!"
+                + " Load the energies levels with SpectrumFactory.load_databank"
+                + "('path', load_energies=True)"
+            ) from err
```

Comment:
If energies haven't been loaded, it should fail on the first key error. We could use your new `get_partition_function_molecule` function to make it clearer.

```suggestion
        parsum = self.get_partition_function_molecule(molecule)[isotope][elec_state]
```
(will fail properly with the right error)

Reply:
Nice Suggestion. Implemented it and updated the error message. Thanks :+1: !